### PR TITLE
RavenDB-14905 Allow the use of unix ticks for time-series bulk-insert

### PR DESCRIPTION
**Pull request by karmeli87**

### CODE REVIEW

**Review by ayende — COMMENTED** [review]
Where is this used?

**Comment by karmeli87** [comment]
This will be used in the Java client, and in other clients as well.
Also needed for the tsbs benchmark, which is written in Go

**Review by ayende — APPROVED** [review]

**Comment by ppekrol** [comment]
@karmeli87 can we write a simple test for this?

**Review by ppekrol — APPROVED** [review]

File: src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs [hunk 1]
```diff
@@ -64,11 +64,27 @@ internal static TimeSeriesOperation Parse(BlittableJsonReaderObject input)
             return result;
         }
 
+        private const string TimeSeriesFormat = "TimeFormat";
+        private const string UnixTimeFormatMs = "UnixTimeFormatMs";
+        private const string UnixTimeFormatNs = "UnixTimeFormatNs";
+        private static readonly long _epochTicks = new DateTime(1970, 1, 1).Ticks;
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs]: Just to be sure, this shouldn't be `new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc).Ticks` ?

File: src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs [hunk 2]
```diff
@@ -64,11 +64,27 @@ internal static TimeSeriesOperation Parse(BlittableJsonReaderObject input)
...
+        {
+            return unixNs / 100 + _epochTicks;
+        }
+
         internal static TimeSeriesOperation ParseForBulkInsert(BlittableJsonReaderObject input)
         {
             if (input.TryGet(nameof(Name), out string name) == false || name == null)
                 ThrowMissingProperty(nameof(Name));
 
+            input.TryGet(TimeSeriesFormat, out string format);
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs]: Hmm... I would prefer to convert this value to enum and if it exists parse it. If it cannot be parsed then we should throw.

File: src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs [hunk 3]
```diff
@@ -64,11 +64,27 @@ internal static TimeSeriesOperation Parse(BlittableJsonReaderObject input)
             return result;
         }
 
+        private const string TimeSeriesFormat = "TimeFormat";
+        private const string UnixTimeFormatMs = "UnixTimeFormatMs";
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs]: UnixTimeInMs?

File: src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs [hunk 4]
```diff
@@ -64,11 +64,27 @@ internal static TimeSeriesOperation Parse(BlittableJsonReaderObject input)
...
+        private const string UnixTimeFormatNs = "UnixTimeFormatNs";
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs]: UnixTimeInNs?

File: src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs [hunk 5]
```diff
@@ -87,9 +103,21 @@ internal static TimeSeriesOperation ParseForBulkInsert(BlittableJsonReaderObject
                     return null; //never hit
                 }
 
+                long time = (long)bjro[0];
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs]: This cannot be LazyNumberValue?

Comment (karmeli87) [reply src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs]: LazyNumberValue has an implicit cast to `long`

File: src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs (hunk 1 above)
Comment (karmeli87) [reply src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs]: Let's be on the safe side, done.

File: src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs (hunk 2 above)
Comment (karmeli87) [reply src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs]: done

File: src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs (hunk 5 above)
Comment (ayende) [reply src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs]: But that won't trigger here. That is something that the _compiler_ inserts, not runtime.

File: src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs [hunk 6]
```diff
@@ -133,6 +171,16 @@ internal static TimeSeriesOperation ParseForBulkInsert(BlittableJsonReaderObject
             result.Appends = new List<AppendOperation>(sorted.Values);
 
             return result;
+
+            static long GetLong(object value)
```
Comment (karmeli87) [inline src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs:175]: Will this box? should we `Inline` this?

Comment (ayende) [reply src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs:175]: No, this shouldn't cause boxing. Why do you expect it to?

**Pull request merged**